sds_data_manager/lambda_code/SDSCode/pipeline_lambdas/dependency.py

```diff
@@ -578,44 +576,116 @@ def get_latest_repoint_file(end_date: datetime) -> Optional[str]:
 
-    # Sort by end_date and version
-    latest = sorted(repoint_files, key=lambda x: (x[0], x[1]))[-1]
-    latest_file_date = latest[0]
+    if not latest_repoint_file:
+        raise ValueError("No Repoint file found in the database.")
 
-    # Check that input end is within latest repoint file end date
-    if latest_file_date < end_date:
+    if latest_repoint_file.end_date < end_date:
         logger.info(
-            f"Latest repoint file end date {latest_file_date} "
+            f"Latest repoint file end date {latest_repoint_file.end_date} "
             f"is before input end date {end_date}"
         )
         return None
 
-    # Otherwise, return the latest repoint file without the path prefix
-    return basename(latest[2])
+    return basename(latest_repoint_file.file_path)
+
+
+def check_requested_kernels(combined_kernel_sources, metakernel_files):
+    """Check if all requested kernels are present in the metakernel files.
+
+    We need to ensure that the returned list of metakernel files includes
+    all requested kernels, especially for ephemeris kernels. The API can
+    return the "best" ephemeris kernels, which can include both historical
+    and predicted kernels depending on the input time range. If the user
+    specifically requests only historical ephemeris kernels, we must verify
+    that only historical files are returned. Otherwise, both historical
+    and predicted kernels are acceptable.
+
+    Additionally, the API can return multiple kernels for the same source
+    if the files cover specific date ranges. Because of this, we must
+    check that all requested sources are present in the returned
+    metakernel files, rather than performing a direct one-to-one
+    comparison. Each source may correspond to multiple kernel files.
+
+    Parameters
+    ----------
+    combined_kernel_sources : str
+        Comma-separated string of requested kernel sources.
+    metakernel_files : list
+        List of metakernel files found.
+
+    Returns
+    -------
+    bool
+        True if all requested kernels are found, False otherwise.
+    """
+    requested_kernels = set(combined_kernel_sources.split(","))
+    expected_ephemeris = set(
+        [kernel for kernel in requested_kernels if "ephemeris_" in kernel]
+    )
+    expected_other_kernels = set(
+        [kernel for kernel in requested_kernels if "ephemeris_" not in kernel]
+    )
+
+    ephemeris_found = set()
+    other_kernels_found = set()
+
+    for file in metakernel_files:
+        file_obj = imap_data_access.SPICEFilePath(file)
+        # Extract the kernel type from the file name
+        kernel_type = file_obj.spice_metadata["type"]
+        if "ephemeris_" in kernel_type:
+            ephemeris_found.add(kernel_type)
+        else:
+            other_kernels_found.add(kernel_type)
+
+    # Check if all other requested kernels are found
+    if expected_other_kernels == other_kernels_found:
+        pass
+    else:
```

Comment:
Invert the condition
```suggestion
    if expected_other_kernels != other_kernels_found:
```